Sources/HummingbirdCore/Server/ServerChildChannel.swift

```diff
@@ -16,9 +16,15 @@ import Logging
 import NIOCore
 import ServiceLifecycle
 
-/// HTTPServer child channel setup protocol
+/// Protocol for typed server child channel
+public protocol ServerChildChannelValue: Sendable {
+    /// Child channel that spawned child channel
+    var channel: Channel { get }
```

Comment:
Isn't it better if we add `var eventLoop: EventLoop` here? This assumes we're using a `NIO.Channel`, whereas we could have other implementations such as Lambda that aren't connected to a socket at all.

I would even consider removing this requirement entirely, and replacing it with something like this:

```swift
func addTask<Failure: Error>(
  _ operation: @escaping () async throws -> Void,
  to taskGroup: inout ThrowingDiscardingTaskGroup<Failure>
)
```

Reply:
The `ServerChildChannel` assumes we are using a NIO.Channel anyway. It is there to setup a NIO child channel. This code is never run on lambda.

I'm not exactly sure how you would use the `addTask` function